vertx-config/src/main/java/io/vertx/config/spi/ConfigStore.java

```diff
@@ -52,29 +53,15 @@ default void get(Handler<AsyncResult<Buffer>> completionHandler) {
 
   /**
    * Retrieves the configuration store in this store.
+   *
+   * @return a {@code Future} of the asynchronous result.
    */
-  default Future<Buffer> get() {
-    ContextInternal context = (ContextInternal) Vertx.currentContext();
-    /*
-     We don't have access to the Vert.x instance here.
-     So, for the transition period, we must be prepared in the rare cases this is not invoked on a Vert.x thread
-     */
-    Promise<Buffer> promise = context != null ? context.promise() : Promise.promise();
-    get(promise);
-    return promise.future();
-  }
+  Future<Buffer> get();
```

Comment:
Please note that in this PR, it removed the `default` implementation of `get()` method which delegated to the callback and it failed at last anyway.

The same for the `close()` method,  and the `process` method in ConfigProsessor.

If this is not what is wanted, please comment and I can roll it back :)